index.js

```diff
@@ -203,7 +203,9 @@
       return requirements[_name] === Constructor ?
         function(typeRep) {
           return funcPath([name], typeRep) ||
-                 implPath([/function (\w*)/.exec(typeRep)[1], name]);
+                 implPath(typeof typeRep === 'function' ?
+                          [/function (\w*)/.exec(typeRep)[1], name] :
+                          [name]);
```

Comment:
I don't think this is correct. We're expecting something like `implPath(['Array', 'fantasy-land/of'])` which is roughly equivalent to `implementations['Array']['fantasy-land/of']`. If the type representative is not a function it doesn't make sense to attempt something like `implementations['fantasy-land/of']`.

Also, the `typeof` check does not prevent `/function (\w*)/.exec(x => x)[1]` from throwing.

I think we want something like this:

```javascript
var f = funcPath([name], typeRep);
if (f == null && typeof typeRep === 'function') {
  var match = /function (\w*)/.exec(typeRep);
  if (match != null) f = implPath([match[1], name]);
}
return f;
```